docs/source/_developers/primer.md

```diff
@@ -0,0 +1,234 @@
+- **Error Handling**: If relevant, show how to handle errors and exceptions.
+- **Consistency**: Ensure your code examples follow the same coding style, including indentation and naming conventions.
+- **GitHub**: Reference the repo if snippets from another RC GitHub project is referred.
+
+##  Language and Terminology
+Clear and consistent language and terminology are crucial for effective communication. Here are some guidelines:
+- **Consistency**: Use terminology consistently. Don't use different terms for the same concept.
+- **Acronyms**: Always define acronyms the first time you use them.
+- **Jargon**: Avoid using jargon unless necessary. If you have to use it, define it first.
+- **Language**: Use either American or British English consistently throughout your documentation.
+
+## Documentation Structure
+- Organize documentation into folders that represent sections, with each file as an item of its section.
+- Use a clear hierarchy for headings and subheadings.
+   - A single level-one heading (i.e., `# HEADING`) at the top of the page: `HEADING` will appear in the table of contents.
+   - Add file path to [index.md] as follows:
+    :::{code snippet} md
+    ```{toctree}
+    :hidden:
+    :caption: <section>
+    
+    <section-directory>/<filename>
+    ```
+    :::
+    Where _<section>_ is the top-level index listed in the table of contents and <filename> is the markdown file. Note that `.md` can be omitted. Furthermore, when naming a file or directory, use all lowercase with no spaces. For example, {ref}`home-directory-storage-quota` is simply named `homequota.md`.
+
+- Ensure each page has a clear purpose and covers a single topic.
+
+##  Links and References
+Links and references are crucial for helping users navigate your content and providing additional context. Here's how you can effectively use them in your documentation:
+- Use descriptive link text instead of generic phrases like "click here" or "learn more."
+- Regularly check for broken links (see *Precommits*).
+- Reference external sources when necessary and provide proper attribution.
+- Use cross-references to link to other sections within the documentation.
+- Descriptive Links: Avoid using phrases like "click here" for hyperlink text. Instead, use descriptive text that informs the reader about the content they'll find when they follow the link.
+
+
+:::{list-table}
+:header-rows: 1
+:widths: 50 50
+:align: left
+
+* - ❌ Do Not
+  - ✅ Do
+* - Click here for more information about the installation process.
+  - Read our comprehensive installation guide.
+:::
+
+- Check Links: Regularly verify that all external links in your documentation are valid and update any changes.
+- Cross-References: Use cross-references to link to other sections of your documentation. This can help readers find related content quickly.
+- Attribution: If you're citing external sources or using someone else's work (e.g., images, code snippets), attribute the source correctly.
```

Comment:
What guide should we use to attribute work properly? Example would be good as well for formatting purposes.

Reply:
I believe we should use bibtex, but this might be a 2.0 topic. I noted it and will return (this has actually been a thought for a bit now, so I am glad you brought it up)

See https://mystmd.org/guide/citations